dm/master/openapi_test.go

```diff
@@ -131,10 +142,11 @@ func (t *openAPISuite) TestRedirectRequestToLeader(c *check.C) {
 	// list source not from leader will get a redirect
 	result2 := testutil.NewRequest().Get(baseURL).Go(t.testT, s2.echo)
 	c.Assert(result2.Code(), check.Equals, http.StatusTemporaryRedirect)
+	cancel()
```

Comment:
this equals to `defer cancel()` above

Reply:
It's not actually the same, because it will trigger defer server.Close() first, goroutine will waitting the `	s.bgFunWg.Wait()`,so the tese will never stop 😂